Solutions/CohesitySecurity/Data Connectors/Helios2Sentinel/IncidentProducer/IncidentProducer.cs

```diff
@@ -0,0 +1,252 @@
+                        output.properties.severity = "High";
+                    else if (sev < 30)
+                        output.properties.severity = "Low";
+                    else
+                        output.properties.severity = "Medium";
+                    break;
+                }
+                i++;
+                if (i == 13) break;
+            }
+            output.properties.Description = alert.alertDocument.alertDescription + ". Alert cause: " + alert.alertDocument.alertCause + ". Anomaly Strength: " + sev + ". Additional Info: " + alert.alertDocument.alertHelpText + ". Helios ID: " + alert.id;
+
+            lock (queueLock)
+            {
+                outputQueueItem.Add(JsonConvert.SerializeObject(output));
+            }
+        }
+
+        private static string GetData(string path)
+        {
+            if (CloudStorageAccount.TryParse(blobStorageConnectionString, out var storageAccount))
+            {
+                // Create the CloudBlobClient that represents the Blob storage endpoint for the storage account.
+                var cloudBlobClient = storageAccount.CreateCloudBlobClient();
+                var container = cloudBlobClient.GetContainerReference(containerName);
+                var blobRef = container.GetBlockBlobReference(path);
+
+                using (var ms = new MemoryStream())
+                    using (var sr = new StreamReader(ms))
+                    {
+                        var task = blobRef.DownloadToStreamAsync(ms);
+                        task.Wait();
+                        ms.Position = 0;
+                        return sr.ReadToEnd();
+                    }
+            }
+            else
+            {
+                throw new InvalidProgramException("Invalid format string");
+            }
+        }
+
+        private static void WriteData(string path, string data, ILogger log)
+        {
+            try
+            {
+                if (CloudStorageAccount.TryParse(blobStorageConnectionString, out var storageAccount))
+                {
+                    var cloudBlobClient = storageAccount.CreateCloudBlobClient();
+                    var container = cloudBlobClient.GetContainerReference(containerName);
+                    var blobRef = container.GetBlockBlobReference(path);
+                    var dataAsBytes = Encoding.UTF8.GetBytes(data);
+                    blobRef.UploadFromByteArrayAsync(dataAsBytes, 0, dataAsBytes.Length).Wait();
+                }
+                else
+                {
+                    throw new InvalidProgramException("Invalid format string");
+                }
+            }
+            catch (Exception ex)
+            {
+                log.LogError("Exception --> " + ex.Message);
+            }
+        }
+
+        private static void TextWriter(string incidentID, string param, string value, ILogger log)
+        {
+            WriteData(incidentID + "\\" + param, value, log);
```

Comment:
It doesn't make sense to have a one-line function, TextWriter. Call WriteData directly.